support-frontend/test/utils/CheckoutValidationRulesTest.scala

```diff
@@ -498,8 +515,8 @@ class DigitalPackValidationTest extends AnyFlatSpec with Matchers {
   }
 
   it should "fail if the source payment field received is an empty string" in {
-    val requestMissingState = validDigitalPackRequest.copy(paymentFields = Left(StripeSourcePaymentFields("", None)))
-    DigitalPackValidation.passes(requestMissingState, monthlyDirectUSDProduct) shouldBe an[Invalid]
+    val requestMissingState = PaymentMethodId("")
+    requestMissingState shouldBe None
```

Comment:
for the PaymentMethodId case class I have made it so you can't even construct one if it's not passing validation.  I.e. Parse don't validate.  It would be nice to move all the validation over to a similar parsing process, so that the type checker proves that it's been validated rather than the tests.